hamilton/graph_types.py

```diff
@@ -28,6 +30,20 @@ class HamiltonNode:
     required_dependencies: typing.Set[str]
     optional_dependencies: typing.Set[str]
 
+    def as_dict(self):
```

Comment:
`dataclasses.as_dict` works  as well, but given that we handle other serialization pieces, I think this is better.

Reply:
In the past I have had success with using [Pydantic](https://docs.pydantic.dev/latest/concepts/serialization/#custom-serializers) to define custom serialization for an individual field. However I don't think it is worth adding adding a dependency for just this use case.